desc/compute/_mercier.py

```diff
@@ -0,0 +1,510 @@
+            / (64 * jnp.pi ** 6)
+            * dp_dpsi
+            * (
+                jnp.sign(data["psi"]) * d2v_dpsi2
+                - mu_0 * dp_dpsi * surface_integrals(grid, A / Bsq, match_grid=True)
+            )
+            * surface_integrals(grid, A / grad_psi_sq * Bsq, match_grid=True)
+        )
+    return data
+
+
+def compute_DGeod(
+    R_lmn,
+    Z_lmn,
+    L_lmn,
+    i_l,
+    Psi,
+    R_transform,
+    Z_transform,
+    L_transform,
+    iota,
+    data=None,
+):
+    """Compute the Mercier stability criterion geodesic curvature term.
+
+    See equation 4.20 in
+    Landreman, M., & Jorge, R. (2020). Magnetic well and Mercier stability of
+    stellarators near the magnetic axis. Journal of Plasma Physics, 86(5), 905860510.
+    doi:10.1017/S002237782000121X.
+
+    Parameters
+    ----------
+    R_lmn : ndarray
+        Spectral coefficients of R(rho,theta,zeta) -- flux surface R coordinate (m).
+    Z_lmn : ndarray
+        Spectral coefficients of Z(rho,theta,zeta) -- flux surface Z coordinate (m).
+    L_lmn : ndarray
+        Spectral coefficients of lambda(rho,theta,zeta) -- poloidal stream function.
+    i_l : ndarray
+        Spectral coefficients of iota(rho) -- rotational transform profile.
+    Psi : float
+        Total toroidal magnetic flux within the last closed flux surface (Wb).
+    R_transform : Transform
+        Transforms R_lmn coefficients to real space.
+    Z_transform : Transform
+        Transforms Z_lmn coefficients to real space.
+    L_transform : Transform
+        Transforms L_lmn coefficients to real space.
+    iota : Profile
+        Transforms i_l coefficients to real space.
+
+    Returns
+    -------
+    data : dict
+        Dictionary of ndarray, shape(num_nodes,) of Mercier criterion geodesic curvature term.
+    """
+    if check_derivs("DGeod", R_transform, Z_transform, L_transform):
+        grid = R_transform.grid
+        data = compute_contravariant_metric_coefficients(
+            R_lmn, Z_lmn, R_transform, Z_transform, data
+        )
+        data = compute_contravariant_current_density(
+            R_lmn,
+            Z_lmn,
+            L_lmn,
+            i_l,
+            Psi,
+            R_transform,
+            Z_transform,
+            L_transform,
+            iota,
+            data,
+        )
+
+        # grad(psi) = grad(rho) * dpsi/drho
+        # A * dtdz = |ds / grad(psi)^3| = |sqrt(g) * grad(rho) / grad(psi)^3| * dtdz
+        A = jnp.abs(data["sqrt(g)"] / data["psi_r"] ** 3) / data["g^rr"]
+        j_dot_b = mu_0 * dot(data["J"], data["B"])
+        Bsq = dot(data["B"], data["B"])
+
+        data["DGeod"] = (
+            expand(
+                grid,
+                jnp.square(surface_integrals(grid, A * j_dot_b))
+                - surface_integrals(grid, A * Bsq)
+                * surface_integrals(grid, A * jnp.square(j_dot_b) / Bsq),
+            )
+            / (64 * jnp.pi ** 6)
+        )
+    return data
+
+
+def compute_AltWell(
```

Comment:
is there a better name than altwell?

Reply:
I haven't seen this form of the magnetic well named anything other than magnetic well in the literature. I thought it might be useful to keep it in since the other magnetic well parameter gives 0 when pressure is 0. If anyone thinks of a better name just reply to this.